### PR TITLE
ScenarioSets use facets, GUI functionality

### PR DESCRIPTION
**Pull request by willu47**
Outstanding:
- [x] An [error occurs](https://www.pivotaltracker.com/story/show/155816758) when saving a scenario set

Numerous improvements to GUI functionality including:
* facets describe the dimensions of a scenario set and all contained scenarios must populate these facets (this is a non-backward compatible change)
* Warning added to prevent deletion of dependent elements (e.g. you cannot delete a sector model used in a sos model unless the sos model configuration is edited first)
* Updated sample project configuration to reflect new scenario set facet format:
```yaml
scenario_sets:
- description: Growth in UK population
  facets:
  - description: null
    name: population
```
and for scenario data:
```yaml
scenarios:
- name: Central Population (Low)
  description: ''
  facets:
  - name: population
    filename: population_low.csv
    spatial_resolution: national
    temporal_resolution: annual
    units: people
  scenario_set: population
- name: Central Population (Medium)
  description: ''
  facets:
  - name: population
    filename: population_med.csv
    spatial_resolution: national
    temporal_resolution: annual
    units: people
  scenario_set: population
```
To recap, each names scenario must have an entry for the facets belonging to a scenario set of which it is part.

### CODE REVIEW

**Comment by RoaldL** [comment]
#155816758 (Error raised by datafileinterface when saving a ScenarioSet) has been resolved. Including a lot of other issues which were caused by asynchronous API calls. Please try again.

File: src/smif/app/src/actions/actions.js [hunk 1]
```diff
@@ -291,6 +294,9 @@ export function deleteSosModel(sosModelName){
                 response => response.json(),
                 error => console.log('An error occurred.', error)
             )
+            .then(
+                date => dispatch(fetchSosModels())
```
Comment (willu47) [inline src/smif/app/src/actions/actions.js]: data?
